mapper-runtime/src/main/java/com/datastax/oss/driver/api/mapper/result/MapperResultProducerService.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.api.mapper.result;
+
+/**
+ * Provides the custom mapper result types that will be used in an application.
+ *
+ * <p>This class is loaded with the Java Service Provider Interface mechanism, you must reference it
+ * via a service descriptor: create a file {@code
+ * META-INF/services/com.datastax.oss.driver.api.mapper.result.MapperResultProducerService}, with a
+ * single line that contains the name of the implementing class.
+ */
+public interface MapperResultProducerService {
```

Comment:
See my other comment, not sure if this interface is required.

Reply:
Oh it just occurred to me: is it because you want to have a deterministic order when loading producers? In this case, OK, I'm fine with this interface.